core/src/main/java/com/microsoft/applicationinsights/internal/config/TelemetryConfigurationFactory.java

```diff
@@ -128,6 +133,51 @@ private void setInternalLogger(ConfigFileParser parser, TelemetryConfiguration c
         InternalLogger.INSTANCE.initialize(loggerOutput, loggerData.items);
     }
 
+    private List<TelemetryModule> setPerformanceCounters(ConfigFileParser parser) {
+        ArrayList<TelemetryModule> modules = new ArrayList<TelemetryModule>();
+
+        ConfigFileParser.StructuredDataResult pcData = parser.getStructuredData(PERFORMANCE_COUNTERS_SECTION, null);
+        if (!pcData.found) {
+            return modules;
+        }
+
+        final ArrayList<String> performanceModuleNames = new ArrayList<String>();
+        TypeReporter reporter = new TypeReporter() {
+            @Override
+            @SuppressWarnings("unchecked")
+            public Class<? extends Annotation>[] annotations() {
+                return new Class[]{PerformanceModule.class};
+            }
+
+            @Override
+            public void reportTypeAnnotation(Class<? extends Annotation> annotation, String className) {
+                performanceModuleNames.add(className);
+            }
+        };
+        final AnnotationDetector cf = new AnnotationDetector(reporter);
+        try {
+            cf.detect("com.microsoft.applicationinsights");
```

Comment:
I think it is better to encapsulate this entire logic of the TypeReporter & AnnotationDetector.
A class with a simple 'detect' API for a given annotation type. The code will be cleaner.